frontend/lib/parsing/parser-error-classes-list.cpp

```diff
@@ -300,6 +302,23 @@ void ErrorDisallowedControlFlow::write(ErrorWriterBase& wr) const {
     }
   }
 
+  // We also special case return statements in special methods that don't allow
+  // returns with values
+  if (auto ret = invalidAst->toReturn()) {
+    if(blockingAst) {
+      if (auto fn = blockingAst->toFunction()) {
+        if((fn->name() == "deinit") || (fn->name() == "postinit")) {
```

Comment:
Is there a reason we need to check the `fn->name()` here? I imagine we want this to work the same regardless of the name of the function we're giving this error for, and checking if a function disallows return based on its name should be happening at the site of error emission anyways.